### PR TITLE
chore: dependency wrangling

### PR DESCRIPTION
**Pull request by pauldambra**
# 1. I got fed up of seeing these warning

```
 WARN  deprecated @babel/plugin-proposal-class-properties@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-class-properties instead.
 WARN  deprecated @babel/plugin-proposal-nullish-coalescing-operator@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-nullish-coalescing-operator instead.
 WARN  deprecated @babel/plugin-proposal-private-property-in-object@7.21.11: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-private-property-in-object instead.
```
so I obeyed them

# 2. turns out we have an eslint plugin for our jest tests but it wasn't actually running anything

so let's make it run

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -60,7 +58,6 @@ module.exports = {
         'react/jsx-curly-brace-presence': ['error', { props: 'never', children: 'never', propElementValues: 'always' }],
         'no-console': ['error', { allow: ['warn', 'error'] }],
         'no-debugger': 'error',
-        'no-only-tests/no-only-tests': 'error',
```
Comment (pauldambra) [inline .eslintrc.js:63]: built in to the jest plugin if you're actually using it

File: frontend/src/queries/nodes/DataNode/dataNodeLogic.queryCancellation.test.ts [hunk 2]
```diff
@@ -46,39 +46,6 @@ describe('dataNodeLogic - query cancellation', () => {
     })
     afterEach(() => logic?.unmount())
 
-    it.skip('cancels a running query', async () => {
```
Comment (pauldambra) [inline frontend/src/queries/nodes/DataNode/dataNodeLogic.queryCancellation.test.ts:49]: it's been skipped for a year, we don't need it

File: frontend/src/scenes/funnels/funnelDataLogic.test.ts [hunk 3]
```diff
@@ -166,13 +166,6 @@ describe('funnelDataLogic', () => {
 
     describe('based on insightDataLogic', () => {
         describe('results', () => {
-            it.skip('with non-funnel insight', async () => {
```
Comment (pauldambra) [inline frontend/src/scenes/funnels/funnelDataLogic.test.ts:169]: it's been skipped for a year, we don't need it

File: frontend/src/scenes/insights/InsightNav/insightNavLogic.test.ts [hunk 4]
```diff
@@ -245,29 +245,6 @@ describe('insightNavLogic', () => {
                 })
             })
 
-            // it('stores series from retention entities', async () => {
```
Comment (pauldambra) [inline frontend/src/scenes/insights/InsightNav/insightNavLogic.test.ts:248]: it's been skipped for a year, we don't need it

File: frontend/src/scenes/insights/insightLogic.test.ts [hunk 5]
```diff
@@ -304,7 +302,7 @@ describe('insightLogic', () => {
         })
     })
 
-    describe.skip('as dashboard item', () => {
```
Comment (pauldambra) [inline frontend/src/scenes/insights/insightLogic.test.ts:307]: it's been skipped for a year, only keep the tests that currently pass when unskipped

File: frontend/src/scenes/insights/insightSceneLogic.test.ts [hunk 6]
```diff
@@ -42,8 +42,7 @@ describe('insightSceneLogic', () => {
             })
     })
 
-    // for some reason this test fails, while the tested behaviour is correct
-    it.skip('redirects when opening /insight/new with filters', async () => {
```
Comment (pauldambra) [inline frontend/src/scenes/insights/insightSceneLogic.test.ts:46]: it doesn't fail for me 🤷

File: frontend/src/scenes/retention/retentionLineGraphLogic.test.ts
Comment (pauldambra) [inline frontend/src/scenes/retention/retentionLineGraphLogic.test.ts:1]: it's been skipped for a year, we don't need it

File: frontend/src/scenes/retention/retentionTableLogic.test.ts
Comment (pauldambra) [inline frontend/src/scenes/retention/retentionTableLogic.test.ts:1]: it's been skipped for a year, we don't need it

File: frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.test.ts [hunk 7]
```diff
@@ -378,47 +377,6 @@ describe('sessionRecordingDataLogic', () => {
                 'loadSnapshotsForSourceSuccess',
             ])
         })
-
-        // regularly times out in CI, let's skip for now since we know this works 🙈
-        it.skip('polls up to a max threshold', async () => {
```
Comment (pauldambra) [inline frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.test.ts:383]: delete not skip!

File: package.json [hunk 8]
```diff
@@ -243,18 +243,17 @@
...
         "concurrently": "^5.3.0",
         "css-loader": "^3.4.2",
-        "cypress": "^13.3.0",
+        "cypress": "^13.11.0",
         "cypress-axe": "^1.5.0",
-        "cypress-terminal-report": "^5.3.7",
+        "cypress-terminal-report": "^6.1.0",
         "esbuild-visualizer": "^0.6.0",
-        "eslint": "^8.52.0",
+        "eslint": "^8.57.0",
```
Comment (pauldambra) [inline package.json:250]: not every dependency we use yet supports eslint 9, so don't go there

**Review by daibhin — APPROVED** [review]

**Pull request merged**